src/main/java/org/radarcns/stream/KafkaStreamFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 King's College London and The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.radarcns.stream;
+
+import org.radarcns.config.RadarBackendOptions;
+import org.radarcns.config.RadarPropertyHandler;
+import org.radarcns.stream.empatica.E4StreamMaster;
+import org.radarcns.stream.phone.PhoneStreamMaster;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class KafkaStreamFactory {
+    private final RadarPropertyHandler properties;
+    private final RadarBackendOptions options;
+
+    public KafkaStreamFactory(RadarBackendOptions options,
+                              RadarPropertyHandler properties) {
+        this.options = options;
+        this.properties = properties;
+    }
+
+    public StreamMaster createStreamWorker() throws IOException {
+        String streamType = properties.getRadarProperties().getStreamWorker();
+        if (streamType == null) {
+            // Try to get the stream type from the commandline arguments
+            String[] args = options.getSubCommandArgs();
+            if (args == null || args.length == 0) {
+                streamType = "all";
+            } else {
+                streamType = args[0];
+            }
+        }
+
+        boolean isStandalone = properties.getRadarProperties().isStandalone();
+        switch (streamType) {
+            case "e4":
```

Comment:
Ok, now I see you point, thx. I do not see this you use-case in RADAR, since anything in terms of streams will be in this project. But, I see your point in terms of being flexible for future projects.

For the time being we decide how to manage this problem, let's convert them in constants. I would avoid hardcoded strings that might be duplicated in different classes, and that may cause annoying problems in the future.